src/utils/get-command-spec.ts

```diff
@@ -57,68 +112,58 @@ export type CommandSpec<Input extends CommandInput = EmptyCommandInput> = OmitEx
 		? NonNullable<Input['options']> extends never
 			? ExcludeMe
 			: {
-					[Option in keyof Input['options']]: OmitExcludeMeProperties<{
-						/** A description of this option, to be shown on help screens. */
-						description?: string;
-
-						/** Whether this option also applies to commands farther down in the file tree. Defaults to `false`. */
-						cascades?: true;
+					[Option in keyof Input['options']]: OmitExcludeMeProperties<
+						{
+							/** A description of this option, to be shown on help screens. */
+							description?: string;
+
+							/** Whether this option also applies to commands farther down in the file tree. Defaults to `false`. */
+							cascades?: true;
+
+							/** A single-character that could be used instead of the full option name. */
+							shorthand?: string;
+
+							/** Whether this option must be provided with this command. Defaults to `false`. */
+							required: undefined extends Input['options'][Option] ? ExcludeMe : true;
+
+							/** What type of value should be provided. Invalid values will be rejected. */
+							type: number extends Input['options'][Option]
+								? Input['options'][Option] extends number
+									? 'integer' | 'float'
+									: ExcludeMe
+								: ExcludeMe;
+
+							/** A finite array of acceptable option values or callback providing same. Invalid values will be rejected. */
```

Comment:
The placement of this TSDoc comment causes it not to appear in VS Code's IntelliSense. See:

<img width="789" alt="Screen Shot 2021-07-26 at 5 13 48 PM" src="https://user-images.githubusercontent.com/3850064/127066037-2d74af8a-ce8c-453d-80af-efc6f83bf6dc.png">

_Compare with `required`'s comment:_

<img width="787" alt="Screen Shot 2021-07-26 at 5 14 27 PM" src="https://user-images.githubusercontent.com/3850064/127066094-271cb548-a4c1-4d57-9ccf-1c6c774b06d5.png">

Reply:
I've updated the comment so that it also indicates whether it's optional or required, and what the implication of that is.